src/alert.cpp

```diff
@@ -278,23 +278,36 @@ namespace libtorrent {
 
 	std::string performance_alert::message() const
 	{
-		static char const* const warning_str[] =
+		static const std::unordered_map<performance_warning_t, char const*> warning_str =
```

Comment:
is it really an improvement to use an unordered map here? It seems it would be a lot better to just enforce this mapping is correct with a test that samples a few errors